engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -377,21 +377,36 @@ private static boolean isErrorLevelEnable(FileAppender fileAppender)
    */
   public static boolean addAppender(String name, Properties properties)
   {
-    if (getAppendersNames().contains(name)) {
-      logger.warn("A logger appender with the name '{}' exists. Cannot add a new logger appender with the same name", name);
+    return addAppender(LogManager.getRootLogger(), name, properties);
+  }
+
+  /**
+   * Adds Logger Appender to a specified logger
+   * @param logger Logger to add appender to, if null, use root logger
+   * @param name Appender name
+   * @param properties Appender properties
+   * @return True if the appender has been added successfully
+   */
+  public static boolean addAppender(Logger logger, String name, Properties properties)
+  {
+    if (logger == null) {
```

Comment:
Is this check needed since addAppender with two arguments is already doing this. If somebody is using the three arg method instead of the two arg isn't it reasonable to expect they would pass a non null logger.

Reply:
The check is necessary as `logger` can't be `null` (to avoid NPE). Instead of throwing an exception a better default is to use root logger.

Reply:
Since there is already a method that deals with the case where there is no logger, I think it is ok to throw NPE in case when a null logger is passed to this method.

Reply:
NPE is only necessary when there is no reasonable way to handle `null`. For this method, `null` means root logger.

Reply:
Why would null be root logger?

Reply:
To keep the initial default and make it consistent with two parameters method.

Reply:
It is a subjective choice and don't see the need to do that, npe is valid as well. If there is any other reason then it might make more sense to do that. It is not a big deal and am fine with it, if you want to keep it this way.

Reply:
I don't see a reason to require `logger` not to be `null` and why raising NPE would be a better choice, but if there is one (ease of use, performance, consistency) please provide the reasoning and I'll change it.

Reply:
Mainly because I see a repetitive null check in all the methods and it's not limited to one area. I would consider equating null to a special logger as doing something extra and not the other way round, i.e., is there a good reason to handle and repeat this null check all over the place. Like I said it is not a blocker for me.

Reply:
As all those places are independent public methods of the LoggerUtility class that do not share any common method that they all call, the check similar to what is done on the line 437 for an invalid argument will still be necessary in all places.

Reply:
Yes, but my primary question is why to opt to handle it in the first place when this special handling will need to be done all over the place.

Reply:
For the same reason why there is a check on line 437: it is the best practice to validate incoming arguments instead of raising NPE in random places.

Reply:
You could validate and throw NPE

Reply:
It goes back to the question why validating and raising NPE will be a better approach